command/init.go

```diff
@@ -168,6 +170,62 @@ job "example" {
         }
       }
 
+      # The template block instantiates an instance of consul-template. This
+      # creates a convenient way to ship configuration files that are populated
+      # from Consul data, retrieve secrets from Vault and even ship static
+      # configurations to Nomad tasks.
+      # template {
+        # The data field allows a template to be embedded in the job file.
+        # This is convenient when the template is simple. The example template
+        # creates a redis config file setting the maxclients field by querying a
+        # key in Consul.
+        # data = "maxclients {{key \"service/redis/maxconns@east-aws\"}}"
+
+        # The destination sets where the rendered template will be outputted.
+        # destination = "local/redis.conf"
+
+        # The change mode determines the behavior when the template is updated.
+        # The possible options are:
+        #   - "noop":    Take no action
+        #   - "restart": Restart the task
+        #   - "signal":  Signal the task
+        # change_mode = "signal"
+
+        # Change signal specifies the signal to send the task when the change_mode
+        # is "signal".
+        # change_signal = "SIGUSR1"
+      # }
+
+      # The Vault block allows the task to specify that it requires a Vault token.
```

Comment:
I think we should move `vault` above `template`. It's more common that you would configure vault, then create a template, and the "flow" makes more sense in my opinion.